pkg/plugin/state.go

```diff
@@ -695,6 +695,33 @@ func (e *AutoscaleEnforcer) handleUpdatedScalingBounds(vm *api.VmInfo, unqualifi
 	klog.Infof(fmtString, podName, pod.node.name, cpuVerdict, memVerdict)
 }
 
+func (e *AutoscaleEnforcer) handleNonAutoscalingUsageChange(vm *api.VmInfo, name util.NamespacedName) {
+	e.state.lock.Lock()
+	defer e.state.lock.Unlock()
+
+	klog.Info("Handling updated usage information for non-autoscaling VM pod %v", name)
+
+	pod, ok := e.state.podMap[name]
+	if !ok {
+		klog.Errorf("[autoscale-enforcer] cannot find non-autoscaling VM pod %v to update usage", name)
+		return
+	}
+	node := pod.node
+
+	newMem, newCPU := vm.Using().Mem, vm.Using().VCPU
+	oldMem, oldCPU := pod.memSlots.Reserved, pod.vCPU.Reserved
+
+	node.memSlots.Reserved, node.vCPU.Reserved = newMem, newCPU
+	pod.memSlots.Reserved, pod.vCPU.Reserved = newMem, newCPU
+
+	klog.Infof(
+		"[autoscale-enforcer]: updated VM pod %v resources: vCPU %v -> %v, updated memory: %v -> %v",
```

Comment:
It'd be good to include information for the node as well, e.g. a format string like:

```
updated VM pod %v resources in node %s: vCPU %v -> %v (node %v -> %v), memory %v -> %v (node %v -> %v)
```

**NB:** Having `node %s` in the format string is useful for log filtering (all other handlers do the same), at least until we make things a little more standard. The rest is just so that we can deduce the state of the node from the logs if the internal logic isn't correct.

Reply:
Still wrapping my head around the pod-node-VM hierarchy but this makes sense.